src/main/java/FrameworksDrivers/ChatView.java

```diff
@@ -162,64 +160,13 @@ public void updatePage(Object[] info) {
                 currChatPanel.setBackground(Color.CYAN);
 
                 //iterate through the messages of the current chatroom
-                for (int i = 2; i < chatroomList.size(); i++) {
```

Comment:
I broke down the huge for loop into a bunch of helper functions

Reply:
Very Swag!